dataprofiler/tests/profilers/test_histogram_utils.py

```diff
@@ -0,0 +1,348 @@
+import unittest
+from collections import defaultdict
+from unittest import mock
+
+import numpy as np
+
+import dataprofiler
+from dataprofiler.profilers import NumericStatsMixin, histogram_utils
+
+
+# Mocks for property functions
+def mock_stddev():
+    return 1.0
+
+
+def mock_invalid_stddev():
+    return -1.0
+
+
+def mock_get_percentile(percentile):
+    if percentile == [75]:
+        return 2.0
+    if percentile == [25]:
+        return 1.0
+
+
+def mock_sqrt_return_none(profile):
+    return None
+
+
+def mock_sqrt_return_nan(profile):
+    return float("nan")
+
+
+class TestColumn(NumericStatsMixin):
+    def __init__(self):
+        NumericStatsMixin.__init__(self)
+        self.match_count = 0
+        self.times = defaultdict(float)
+        self.match_count = 5
+        self.min = 1
+        self.max = 5
+        self._biased_skewness = 1.0
+        self._stored_histogram["histogram"]["bin_counts"] = [1, 1, 1, 1, 1, 1]
+        self._stored_histogram["histogram"]["bin_edges"] = [0, 1, 2, 3, 4, 5, 6]
+
+    def update(self, df_series):
+        pass
+
+    def _filter_properties_w_options(self, calculations, options):
+        pass
+
+
+class TestHistogramUtils(unittest.TestCase):
+    def test_ptp(self):
+        # Maximum greater than minimum
+        maximum = 15
+        minimum = 12
+        expected = 3
+        actual = histogram_utils._ptp(maximum, minimum)
+        self.assertEqual(expected, actual)
+
+        # Minimum greater than maximum
+        maximum = 12
+        minimum = 15
+        expected = -3
+        actual = histogram_utils._ptp(maximum, minimum)
+        self.assertEqual(expected, actual)
+
+        # Minimum equal to maximum
+        maximum = 15
+        minimum = 15
+        expected = 0
+        actual = histogram_utils._ptp(maximum, minimum)
+        self.assertEqual(expected, actual)
+
+    def test_calc_doane_bin_width_from_profile(self):
+        # Initial setup of profile
+        profile = TestColumn()
+
+        with mock.patch(
+            "dataprofiler.profilers.NumericStatsMixin.stddev", new_callable=mock_stddev
+        ):
+            # Case 1: min, max, match_count, biased_skewness, and stddev are set
+            sg1 = np.sqrt(6.0 * (5 - 2) / ((5 + 1.0) * (5 + 3)))
```

Comment:
Should we replace the values themselves with the settings above?

e.g. `1-> self.min`